### PR TITLE
First draft

### PR DESCRIPTION
**Pull request by BobJWalker**
# About

The benefits of a isolated tenanted infrastructure

# Release

It is not, it is for the multi-tenancy campaign

# Blog image idea

The image should represent a SaaS multi-tenancy approach with isolated infrastructure.  It should demonstrate each customer has a unique copy of the application.  We can probably re-use some of the work we've done for the new landing pages.

# Prerequisites

- [x] The draft is complete and this post is ready to be reviewed.
- [x] I'm familiar with the [guide for blog content basics](https://www.octopus.design/932c0f1a9/p/901d2a-blog-content) and [writing educational content](https://www.octopus.design/932c0f1a9/p/137b06-educational-content---how-we-write-about-what-we-do).
- [x] I've checked the content using Grammarly (Octopus has a corporate license) and the [Hemingway app](https://hemingwayapp.com/) to remove typos, grammatical errors, and passive voice. 
- [x] The PR build passes validation, and if it doesn't, I've checked the [common validations errors](https://www.octopus.design/932c0f1a9/p/901d2a-blog-content/t/817249) and none of those apply.
- [x] The images I've included follow the [rules for working with images](https://www.octopus.design/932c0f1a9/p/5061d7-working-with-images).
- [x] I've added alt text for all the images I've included (alt text describes an image to people unable to see it, 125 characters max)
- [ ] I've drafted social media posts for Twitter and LinkedIn and will post them for review in #topic-social-media.
- [ ] I've dropped a note in Slack in #topic-content-marketing to let Tegan Ali know my post is ready for review. (No need to fill out a content brief.)

## How to review this PR

Does it match our other articles for multi-tenancy?  I don't mention Octopus Deploy specifically.

## Publication date

N/A

### CODE REVIEW

**Review by steve-fenton-octopus — COMMENTED** [review]
I'm heartily behind this article. The economics are so different now to the early 2000s. If I wrote a new software product offering today, it would 100% be infrastructure isolated, not shared. There are lots of clever ways of making the infrastructure cheaper without causing all those complexity problems.

I have a couple of comments, I hope they are useful.

File: blog/2023-08/benefits-isolated-tenanted-infrastructure/index.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+The likelihood of all customers being on the same version is low.  One group of customers might be on 2023.3.568, while another group is on 2023.2.990.  Most SaaS products have different tiers or features available as add-ons.  
+
+The possible versions and feature sets to support will exponentially grow.  That requires robust testing automation, including the ability to spin up new “test customers” or “test environments” to support all the various permutations.  
+
+### Higher cloud bills
+
+With a shared application and database model, all customers share the same pool of resources.  The chances of all customers needing all the resources at any given time are near zero.
+
+However, the inverse is true with isolated tenanted infrastructure.  Applications that can only run on Virtual Machines will “bursty” resource use.  A lot of time, the VM will be idle, consuming minimal resources.  As such, they will incur higher cloud bills than those that run on platforms that allow sharing of resources with the required isolation.  For example, container platforms such as ACS, ECS, Kubernetes, PaaS platforms such as Azure Web Apps, Lambdas, and Functions, or Azure’s Elastic database pools.
```
[the remark this answers is not in the document]
Comment (BobJWalker) [reply blog/2023-08/benefits-isolated-tenanted-infrastructure/index.md]: Added to my group of changes.

File: blog/2023-08/benefits-isolated-tenanted-infrastructure/index.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+- You have a single code base *without* needing a where clause containing `where customerId=` for every query.
+- You can make an existing application "multi-tenant" with minimal code changes.
+- The customer’s data is isolated from other customers.  Unique database accounts help prevent customers from seeing each other’s data.
+- Isolated databases lower the risk of a noisy customer causing issues with other customers.
+- Every customer is on the same version of the application.  
+
+:::hint A schema per customer is nearly identical to this approach.  The primary difference is that it doesn’t solve the database size, performance, and maintenance issues.  All the issues discussed in this section apply to that approach.
+
+I spent more than 5 years building multi-tenant applications using this approach.  After seeing the good and the bad, I would avoid it at all costs.
```
Comment (TeganAli) [inline blog/2023-08/benefits-isolated-tenanted-infrastructure/index.md:127]: @BobJWalker I added this text to a hint box, but just want to check that these 2 paragraphs definitely belong together in the hint box? 
I think we might need to be more specific in the second sentence that says "I spent more than 5 years building multi-tenant applications using this approach." and say "I spent more than 5 years building multi-tenant applications using a schema per customer approach." if that's what "this approach" is referring to. 
Thanks :)

Comment (BobJWalker) [reply blog/2023-08/benefits-isolated-tenanted-infrastructure/index.md:127]: This should be all that is in the hint box:

> :::hint A schema per customer is nearly identical to this approach.  The primary difference is that it doesn’t solve the database size, performance, and maintenance issues.  All the issues discussed in this section apply to that approach.

Comment (TeganAli) [reply blog/2023-08/benefits-isolated-tenanted-infrastructure/index.md:127]: @BobJWalker thank you for clarifying, I'll fix that up :)

**Review by TeganAli — APPROVED** [review]

**Pull request merged**